draft-ietf-tls-esni.md

```diff
@@ -620,16 +623,24 @@ error code.
 
 ### HelloRetryRequest {#hrr}
 
+As required in {{send-ech}}, clients offering ECH MUST ensure that
+HelloRetryRequest-sensitive parameters applicable to TLS 1.3 and higher
+versions in ClientHelloInner match that in ClientHelloOuter. These parameters
+include:
+
+1. TLS 1.3 {{!RFC8446}} ciphersuites in the ClientHello.cipher_suites list; and
+1. The "key_share" and "supported_groups" extensions {{RFC8446}}. (These
+extensions may be copied from ClientHelloOuter into ClientHelloInner as
+described in {{send-ech}}.)
```

Comment:
I added "supported_versions" to the list!

Reply:
Sorry, I should have been clearer. So the reason supported_versions is a huge headache is that it *doesn't* match between the inner and outer. The inner ClientHello is forbidden from offering TLS 1.2, while the outer one may still offer it. I think the rule is something like they need to match, after you drop 1.0-1.2 from the outer list, which is weird.

Reply:
Yeah, I figured this was covered by saying these are "parameters applicable to TLS 1.3 or higher". Do you not think that's enough? If not, would you mind proposing text to clarify?

Reply:
Ah I see. Yeah, I didn't interpret that to apply to the individual details like that. Do you think this would be clearer?

> 1. TLS 1.3 {{!RFC8446}} ciphersuites in the ClientHello.cipher_suites list
> 1. The "key_share" and "supported_groups" extensions
{{RFC8446}}. (These extensions may be copied from ClientHelloOuter into
ClientHelloInner as described in {{send-ech}}.)
> 1. Versions in the "supported_versions" extension, excluding TLS 1.2 and earlier. Note the ClientHelloOuter MAY include these older versions, while the ClientHelloInner MUST omit them.

Reply:
Yep, that is more clear!